secrets/secrets.go

```diff
@@ -15,6 +15,14 @@ const (
 	credentialSecret = "credential"
 )
 
+// A Secret is the base type of secrets.
+type Secret []byte
+
+// IsEmpty returns true if the secret is empty.
+func (s Secret) IsEmpty() bool {
+	return len(s) == 0
```

Comment:
should we check if it's nil as well?

Reply:
nil slice is empty slice (`len(nil)` will return 0).

Reply:
nope we don't need to :)